Add dock margin size customization feature

mojoro opened the pull request:
### Description

I added a feature to customize the dock margin size when panel mode is enabled. This feature introduces a slider in the settings, allowing users to adjust the margin from the edge of the screen (default: 0px) up to 300px.

This is particularly useful for setups with differently sized monitors, where getting a consistent amount of space from the edge of the screen can be difficult without access to dynamic units (e.g. vw, %, etc.)
![dock-margin-demonstration](https://github.com/user-attachments/assets/d7f042bb-35b6-4594-8583-3f119146782b)

### Implementation Details

Added a new GSettings key ([dock-margin-size](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)) to store the margin value.
Updated the settings UI ([Settings.ui](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)) to include a slider for margin adjustment.
Modified the dock positioning logic in [docking.js](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) to respect the margin value.
Bound the slider to the GSettings key in [prefs.js](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html).

### Testing

Tested on GNOME Shell 46 with a dual-monitor setup. (1920x1080 and 3440x1440)
Verified that the margin applies correctly in panel mode, both vertically and horizontally.
Confirmed that the default behavior (0px margin) remains unchanged.

#### Sidenote
This is my first contribution to anything open-source, so feedback is very welcome. If you'd like me to change anything, please let me know.

mojoro committed:
Add dock margin size customization feature

mojoro committed:
Revert mistaken deletion of line in Settings.ui

3v1n0 reviewed (commented):
Thanks looks good and code changes are small enough not to be a problem to maintain, but... Shouldn't the dock borders also be rounded in such case?

I feel the squared borders don't fit well with this case... So I guess we need some new style classes

mojoro committed:
Use rounded borders when dock margin is enabled

mojoro commented:
Good observation. I didn't notice that. I updated it to use rounded margins when the margin length is greater than 0, and I removed that stray new-line you called out. Sorry it took me so long to get to!

![dock-margin-rounded-borders](https://github.com/user-attachments/assets/748ce375-afcc-4729-8c92-da413dd2e7f9)

3v1n0 commented on prefs.js:
```diff
@@ -285,6 +285,14 @@ const DockSettings = GObject.registerClass({
+  
+        // Update the GSettings key
+        this._settings.set_int("dock-margin-size", value);
+      }
```
This is not needed if there is already the settings binding, no?

3v1n0 commented on prefs.js:
```diff
@@ -587,18 +595,44 @@ const DockSettings = GObject.registerClass({
+            "dock_margin_size_scale"
+        );
+        dockMarginSizeScale.set_range(0, 300); // Set the range for the margin size
+        dockMarginSizeScale.set_value(this._settings.get_int("dock-margin-size")); // Initialize with the current value
```
IIRc this is not needed if the binding is used later.

3v1n0 commented on prefs.js:
```diff
@@ -587,18 +595,44 @@ const DockSettings = GObject.registerClass({
+        dockMarginSizeScale.connect("value-changed", (scale) => {
+            // Update the GSettings key when the value changes
+            this._settings.set_int("dock-margin-size", scale.get_value());
+        });
```
again, this should be managed by the binding

mojoro committed:
Change margin size settings implementation to use Gtk binding

mojoro commented:
Thank you for the swift feedback!

The previous approach was based on what I could surmise from the GNOME documentation and other size options' implementations and limited by my understanding of the Gtk. 

You are right that managing all of that with the binding is much cleaner. This is my first time working with the Gtk settings, so please let me know if the current implementation matches your expectations, or should I have kept the `dock_margin_size_value_changed_cb` and updated it to match more closely the debounce behavior in `dock_size_scale_value_changed_cb` and `icon_size_scale_value_changed_cb`, prefs.js:276-297?
